internal/flag/internal_flag.go

```diff
@@ -68,6 +72,17 @@ func (f *InternalFlag) Value(
 		maps.Copy(evaluationCtx.GetCustom(), flagContext.EvaluationContextEnrichment)
 	}
 
+	key, keyError := f.GetTargetingKey(evaluationCtx)
+
+	if keyError != nil {
+		return flagContext.DefaultSdkValue, ResolutionDetails{
+			Variant:   VariationSDKDefault,
+			Reason:    ReasonError,
+			ErrorCode: ErrorCodeBucketingKeyMissing,
```

Comment:
One thing we can do to be explicit about the error is to add an `ErrorDetails` field in the `ResolutionDetails` object that explains the error.

This is also following what OpenFeature is doing with the error message https://openfeature.dev/docs/reference/concepts/evaluation-api/#evaluation-details-structure-fields